File: C/pub1/Distrib/LIB/LIB-RUBY/Example/read.temp.download.folder.rb
#!ruby.exe
#**********************************************************
#* read.temp.download.folder
#*********************************************************/

require 'tmpdir'
require 'win32/registry'

###
# getDownloadFolder
###
def getDownloadFolder
    shcu_node = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Folders";
    fo_downloads = "{374DE290-123F-4565-9164-39C4925E467B}"
    Win32::Registry::HKEY_CURRENT_USER.open(shcu_node) do |regnode|
        value = regnode[fo_downloads]
        return value
    end
end

###
# getTempEnviron
###
def getTempEnviron
    fo_temp = Dir::tmpdir()
    return fo_temp
end

puts getDownloadFolder()
puts getTempEnviron()

File: C-Source/pub1/Distrib/LIB/LIB-PHP/Example/read.temp.download.folder.php
#!php.exe
<?php
/**********************************************************
 * read.temp.download.folder
**********************************************************/
function getDownloadsFolder() {
    $SHCU_NODE = "HKCU\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Folders\\";
    $FO_Downloads = "{374DE290-123F-4565-9164-39C4925E467B}";
    $SHCU_Downloads = $SHCU_NODE . $FO_Downloads;
    $wscript = new COM('WScript.Shell');
    $a = $wscript->RegRead($SHCU_Downloads);
    if(is_dir($a))
        return $a;
    else
        return "";
}

function getTempFolder(){
    $FO_TEMP = $_SERVER['TEMP'];
    if(is_dir($FO_TEMP))
        return $FO_TEMP;
    else
        return "";
}

echo("\nDownloads Folder = " . getDownloadsFolder());
echo("\nTemprorary Folder = " . getTempFolder());
?>